### PR TITLE
add more `renderToPdf` options

### PR DESCRIPTION
**Pull request by adghayes**
## Overview

This is a preparatory PR for the VxScan Fujitsu integration, consisting of two parts.

First, when we send a tally report to the Fujitsu, I'd like it to be rendered as a single, long page. Rather than multiple letter pages printed out in sequence. Doing it as multiple letter pages means that there are weird gaps between results in each column between "page breaks."

Second, refactor to allow rendering multiple PDFs at once. Most of the performance cost with printing is from the browser startup, so while it's not necessary, it's a nice boost to be able to startup Playwright once and render all necessary reports at once. In VxScan, this could easily be 6 documents (democratic page for print, republican page for print, non-partisan page for print, democratic page for saving as file, republican page for saving as file, non-partisan page for saving as file).

### CODE REVIEW

**Review by jonahkagan — APPROVED** [review]

File: libs/printing/src/render.tsx [hunk 1]
```diff
@@ -123,25 +87,78 @@ export async function renderToPdf(
...
-    : DEFAULT_PDF_HEIGHT_INCHES;
-  const pdfBuffer = await page.pdf({
-    path: options.outputPath,
-    width: inchesToText(DEFAULT_PDF_WIDTH_INCHES),
-    height: inchesToText(heightInches),
-    margin: mapObject(DEFAULT_PDF_MARGIN_INCHES, inchesToText),
-    printBackground: true, // necessary to render shaded backgrounds
-  });
+
+  const buffers: Buffer[] = [];
```
[the remark this answers is not in the document]
Comment (eventualbuddha) [reply libs/printing/src/render.tsx:91]: As part of this you could simplify the signature to always expect `RenderSpec[]` (or, perhaps, `Iterable<RenderSpec>`) and always return `AsyncIterable<Buffer>`.

File: libs/printing/src/render.tsx [hunk 2]
```diff
@@ -10,13 +10,52 @@ import {
...
+        'content-wrapper' // CONTENT_WRAPPER_ID not defined in this scope
+      ) as HTMLElement
+    ).getBoundingClientRect();
+    return rect.height + rect.top;
+  });
+}
+
+interface RenderOptions {
+  outputPath?: string;
+  expandPageToFitContent?: boolean;
```
[the remark this answers is not in the document]
Comment (adghayes) [reply libs/printing/src/render.tsx]: [33e037c](https://github.com/votingworks/vxsuite/pull/4690/commits/33e037c9b61462ffab74a5cd43bdab2b5c56ce8c)

File: libs/printing/src/render.tsx [hunk 3]
```diff
@@ -53,53 +53,17 @@ interface RenderOptions {
   expandPageToFitContent?: boolean;
 }
 
+interface RenderSpec {
+  document: JSX.Element;
+  options?: RenderOptions;
```
[the remark this answers is not in the document]
Comment (adghayes) [reply libs/printing/src/render.tsx]: [33e037c](https://github.com/votingworks/vxsuite/pull/4690/commits/33e037c9b61462ffab74a5cd43bdab2b5c56ce8c)

File: libs/printing/src/render.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (adghayes) [reply libs/printing/src/render.tsx:82]: Makes sense, going to hold on this until I see how exactly I use this in VxScan.

**Pull request merged**